### PR TITLE
Feature/fetch the weather in san francisco

### PR DESCRIPTION
**Pull request by alonshp**

### CODE REVIEW

File: FinalProjectTests/FinalProjectTests.swift [hunk 1]
```diff
@@ -25,6 +29,41 @@ class FinalProjectTests: XCTestCase {
...
+                XCTAssertEqual(weatherData.temperature , "50.8 F (10.4 C)")
+            } else {
+                XCTFail()
+            }
+        } catch {
+            print(error.localizedDescription)
+        }
+    }
+    
+    func parse() -> WeatherData? {
```
Comment (oded-regev) [inline FinalProjectTests/FinalProjectTests.swift]: remove this method - this should be a method in the application project...

File: FinalProjectTests/FinalProjectTests.swift [hunk 2]
```diff
@@ -7,9 +7,13 @@
 //
 
 import XCTest
+@testable import FinalProject
+
 
 class FinalProjectTests: XCTestCase {
     
+    var weatherWuResponse = [String : Any]()
```
Comment (oded-regev) [inline FinalProjectTests/FinalProjectTests.swift]: it doesn't make sense to use a global var for this

File: FinalProjectTests/FinalProjectTests.swift [hunk 3]
```diff
@@ -25,6 +29,41 @@ class FinalProjectTests: XCTestCase {
...
+                let json = try JSONSerialization.jsonObject(with: data, options: [])
+                if let object = json as? [String: Any] {
+                    weatherWuResponse = object
+                } else {
+                    print("JSON is invalid")
+                }
+            } else {
+                print("no file")
+            }
+            if let weatherData = parse() {
```
Comment (oded-regev) [inline FinalProjectTests/FinalProjectTests.swift]: test with application code

File: FinalProject/WeatherData.swift [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+//  Copyright © 2018 Alon Shprung. All rights reserved.
+//
+
+import Foundation
+
+struct WeatherData {
+    var weatherString: String
+    var temperature: String
+    
+    static func parseJsonDictionaryToWeatherData(jsonDictionary: [String : Any]) -> WeatherData? {
```
Comment (oded-regev) [inline FinalProject/WeatherData.swift]: Looks great 👍

File: FinalProject/NetworkUtils.swift [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+//
+//  Created by Alon Shprung on 4/22/18.
+//  Copyright © 2018 Alon Shprung. All rights reserved.
+//
+
+import Foundation
+import Alamofire
+
+
+struct NetworkUtils {
```
Comment (oded-regev) [inline FinalProject/NetworkUtils.swift]: Looks great 👍

File: FinalProjectTests/FinalProjectTests.swift [hunk 6]
```diff
@@ -25,6 +29,41 @@ class FinalProjectTests: XCTestCase {
...
+        if let currentObservation = weatherWuResponse["current_observation"] as? Dictionary<String,Any>,
+            let weatherString = currentObservation["weather"] as? String,
+            let temperature = currentObservation["temperature_string"] as? String {
+                return WeatherData(weatherString: weatherString, temperature: temperature)
+        } else {
+            return nil
+        }
+    }
+    
     func testPerformanceExample() {
```
Comment (oded-regev) [inline FinalProjectTests/FinalProjectTests.swift]: remove unused test...

File: FinalProjectTests/FinalProjectTests.swift [hunk 7]
```diff
@@ -25,6 +29,41 @@ class FinalProjectTests: XCTestCase {
         // Use XCTAssert and related functions to verify your tests produce the correct results.
     }
     
+    func testJsonToDictionary() {
+        do {
+            let testBundle = Bundle(for: type(of: self))
```
Comment (oded-regev) [inline FinalProjectTests/FinalProjectTests.swift]: The "reading json from file" part, should be in "setUp()"

File: FinalProjectTests/FinalProjectTests.swift (hunk 2 above)
Comment (alonshp) [reply FinalProjectTests/FinalProjectTests.swift]: I using global var that will be initialize in setUp()

File: FinalProjectTests/FinalProjectTests.swift [hunk 8]
```diff
@@ -12,7 +12,7 @@ import XCTest
 
 class FinalProjectTests: XCTestCase {
     
-    var weatherWuResponse = [String : Any]()
+    var weatherWuJson: Any?
```
Comment (alonshp) [inline FinalProjectTests/FinalProjectTests.swift]: now only the json is global and the dictionary inside the test function

File: FinalProjectTests/FinalProjectTests.swift [hunk 9]
```diff
@@ -7,29 +7,47 @@
...
-    
-    func testPerformanceExample() {
-        // This is an example of a performance test case.
-        self.measure {
-            // Put the code you want to measure the time of here.
+    func testJsonToDictionary() {
+        var weatherWuResponse = [String : Any]()
+        
+        if let object = weatherWuJson as? [String: Any] {
+            weatherWuResponse = object
```
Comment (oded-regev) [inline FinalProjectTests/FinalProjectTests.swift]: please use `self.weatherWuJson` whenever you use weatherWuJson

**Pull request merged**